### PR TITLE
Shop Benchmark

### PR DESCRIPTION
**Pull request by kraftp**

### CODE REVIEW

**Comment by kraftp** [comment]
Results: https://docs.google.com/spreadsheets/d/1HDuvIWOR_KX2ZF12-GzfSDaD5-YoSu8yZRbnyasfLrc/edit#gid=1639328652

File: src/main/java/org/dbos/apiary/procedures/postgres/shop/ShopGetItem.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class ShopGetItem extends PostgresFunction {
+
+    public static int runFunction(PostgresContext ctxt, int personID, String searchText, int maxCost) throws Exception {
+        String[] items = ctxt.apiaryCallFunction("ShopESSearchItem", searchText, maxCost).getStringArray();
+        if (items.length > 0) {
+            int bestItem = Integer.parseInt(items[0]);
+            ctxt.apiaryCallFunction("ShopAddCart", personID, bestItem);
```
Comment (qianl15) [inline src/main/java/org/dbos/apiary/procedures/postgres/shop/ShopGetItem.java:15]: It looks a bit weird to do a Postgres write inside a seemingly read-only function. I wonder if it is possible to separately investigate latency from Postgres vs. latency from Elasticsearch?

File: src/test/java/org/dbos/apiary/CrossDBBenchmarkTests.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        apiaryWorker = new ApiaryWorker(new ApiaryNaiveScheduler(), 4);
+        apiaryWorker.registerConnection(ApiaryConfig.elasticsearch, conn);
+        apiaryWorker.registerConnection(ApiaryConfig.postgres, pconn);
+        apiaryWorker.registerFunction("ShopAddItem", ApiaryConfig.postgres, ShopAddItem::new);
+        apiaryWorker.registerFunction("ShopSearchItem", ApiaryConfig.postgres, ShopSearchItem::new);
+        apiaryWorker.registerFunction("ShopAddCart", ApiaryConfig.postgres, ShopAddCart::new);
+        apiaryWorker.registerFunction("ShopCheckoutCart", ApiaryConfig.postgres, ShopCheckoutCart::new);
+        apiaryWorker.registerFunction("ShopGetItem", ApiaryConfig.postgres, ShopGetItem::new);
+        apiaryWorker.registerFunction("ShopESAddItem", ApiaryConfig.elasticsearch, ShopESAddItem::new);
+        apiaryWorker.registerFunction("ShopESSearchItem", ApiaryConfig.elasticsearch, ShopESSearchItem::new);
```
Comment (qianl15) [inline src/test/java/org/dbos/apiary/CrossDBBenchmarkTests.java:89]: These lines seem to be the same as the ones in the `ApiaryWorkerExecutable.java` file. Maybe we could just write one "BenchmarkSetup" function in utilities.

**Comment by qianl15** [comment]
New changes look good. Please merge and add a note that we no longer use nested transactions in Postgres.

**Review by qianl15 — APPROVED** [review]

**Pull request merged**